### PR TITLE
backend-plugin-api: nicer docs for core services

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

More work towards #13082, cleaning up the docs for `coreServices` and aligning so that all core services types as declared as interfaces.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — APPROVED** [review]

File: packages/backend-plugin-api/src/services/definitions/coreServices.ts [hunk 1]
```diff
@@ -14,17 +14,165 @@
...
+ * @public
+ */
+export namespace coreServices {
+  /**
+   * The service reference for the plugin scoped {@link CacheService}.
+   *
+   * @public
+   */
+  export const cache = createServiceRef<import('./CacheService').CacheService>({
+    id: 'core.logger',
```
Comment (freben) [inline packages/backend-plugin-api/src/services/definitions/coreServices.ts:31]: cache?

File: packages/backend-plugin-api/src/services/definitions/coreServices.ts [hunk 2]
```diff
@@ -14,17 +14,165 @@
...
+
+  /**
+   * The service reference for the plugin scoped {@link PluginMetadataService}.
+   *
+   * @public
+   */
+  export const pluginMetadata = createServiceRef<
+    import('./PluginMetadataService').PluginMetadataService
+  >({
+    id: 'core.plugin-metadata',
```
Comment (freben) [inline packages/backend-plugin-api/src/services/definitions/coreServices.ts:119]: The others are camel cased - just noting

File: packages/backend-plugin-api/src/services/definitions/coreServices.ts [hunk 3]
```diff
@@ -14,17 +14,165 @@
...
+  /**
+   * The service reference for the plugin scoped {@link LifecycleService}.
+   *
+   * @public
+   */
+  export const lifecycle = createServiceRef<
+    import('./LifecycleService').LifecycleService
+  >({
+    id: 'core.lifecycle',
+    scope: 'plugin',
```
Comment (freben) [inline packages/backend-plugin-api/src/services/definitions/coreServices.ts:86]: Redundant

**Review by freben — COMMENTED** [review]
Nice!

File: packages/backend-plugin-api/src/services/definitions/coreServices.ts (hunk 2 above)
Comment (Rugvip) [reply packages/backend-plugin-api/src/services/definitions/coreServices.ts:119]: Yep wanna fix all of that in a followup

**Pull request merged**